src/compiler/checker.ts

```diff
@@ -10403,6 +10433,7 @@ namespace ts {
                 case SyntaxKind.TemplateExpression:
                     return checkTemplateExpression(<TemplateExpression>node);
                 case SyntaxKind.StringLiteral:
+                    return checkStringLiteralExpression(<LiteralExpression>node);
                 case SyntaxKind.NoSubstitutionTemplateLiteral:

```

Comment:
Quite significant, reasonably 2-3x faster checking of string literals.

If you consider that every string has on average 5 bytes, you're making 1-5 integers checks for every comparison:
https://github.com/Microsoft/TypeScript/pull/5185/files#diff-c3ed224e4daa84352f7f1abcd23e8ccaR10294

If you on the other hand, you take a cost initially to map a string to a ```bit``` in an integer then most of the time you have at most 1 integer comparison (in which there's 31 strings).

There's lots of optimizations you can do by doing this, even map the properties of objects to an ```integer bits```:
https://github.com/Microsoft/TypeScript/blob/master/src/compiler/checker.ts#L5090

Would have to update patch to compare with this approach but quite confident about :rocket: performance.